rosdoc2/verbs/scan/impl.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2024 R. Kent James <kent@caspia.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import multiprocessing as mp
+import os
+import signal
+import sys
+import threading
+import time
+
+from catkin_pkg.packages import find_packages_allowing_duplicates
+from rosdoc2.verbs.build.impl import main_impl as build_main_impl
+from rosdoc2.verbs.build.impl import prepare_arguments as build_prepare_arguments
+
+logging.basicConfig(
+    format='[%(name)s] [%(levelname)s] %(message)s', level=logging.INFO)
+logger_scan = logging.getLogger('rosdoc2.scan')
+
+goptions = None
+# Setting the BATCH_SIZE or MAX_PACKAGES to a smaller value may be useful in debugging
+# this module, to reduce run time or isolate sections that cause hangs.
+BATCH_SIZE = 10000
+MAX_PACKAGES = 10000
+WATCHDOG_TIMEOUT = 15 * 60  # Seconds
+
+
+def main(options):
+    """Execute the program, catching errors."""
+    try:
+        return main_impl(options)
+    except Exception as e:  # noqa: B902
+        if options.debug:
+            raise
+        else:
+            sys.exit(str(e))
+
+
+class Struct:
+    """Wrap argparse options to allow copies."""
+
+    def __init__(self, **entries):
+        """Create a dictionary from option entries."""
+        self.__dict__.update(entries)
+
+
+def prepare_arguments(parser):
```

Comment:
It would be good to expose BATCH_SIZE here too otherwise it's a magic number.

Reply:
As discussed below, I don't think this is useful to the typical user, so I'll just remove batching instead.